src/controllers/movies.controller.spec.ts

```diff
@@ -0,0 +1,226 @@
+import { Request, Response } from "express";
+import { MockProxy, mock } from "jest-mock-extended";
+
+import { IMovieDocument } from "./../models";
+import { MoviesService } from "./../services/movies.service";
+import { MoviesController } from "./movies.controller";
+
+describe("MoviesController", () => {
+  let moviesController: MoviesController;
+  let moviesService: MockProxy<MoviesService>;
+
+  beforeEach(() => {
+    moviesService = mock<MoviesService>();
+    moviesController = new MoviesController(moviesService);
+  });
+
+  it("should get movie by ID", async () => {
+    const mockRequest = { params: { id: "mockId" } } as unknown as Request;
+    const expectedResult = {
+      _id: "mockId",
+      title: "New Mock Movie",
+      description: "description",
+      releaseDate: "2024-07-20",
+      genre: ["comedy"],
+    } as unknown as IMovieDocument;
+    moviesService.findById.mockResolvedValue(expectedResult);
+
+    const actualResult = await moviesController.getMovie(mockRequest);
+
+    expect(actualResult).toBe(expectedResult);
+  });
+
+  it("should create a new movie", async () => {
+    const mockRequest = {
+      body: {
+        title: "New Mock Movie",
+        description: "description",
+        releaseDate: "2024-07-20",
+        genre: ["comedy"],
+      },
+    } as Request;
+    const expectedResult = {
+      _id: "mockId",
+      title: "New Mock Movie",
+      description: "description",
+      releaseDate: "2024-07-20",
+      genre: ["comedy"],
+    } as unknown as IMovieDocument;
+    moviesService.createMovie.mockResolvedValue(expectedResult);
+
+    const actualResult = await moviesController.createMovie(mockRequest);
+
+    expect(actualResult).toBe(expectedResult);
+  });
+
+  it("should update an existing movie", async () => {
+    const mockRequest = {
+      params: { id: "mockId" },
+      body: {
+        title: "Update Movie",
+        description: "description",
+        releaseDate: "2024-07-20",
+        genre: ["comedy"],
+      },
+    } as unknown as Request;
+    const expectedResult = {
+      _id: "mockId",
+      title: "Update Movie",
+      description: "description",
+      releaseDate: "2024-07-20",
+      genre: ["comedy"],
```

Comment:
Same as above. This mocks are very similar. The content of the test fields is not essential here. You can create a universal mock and reuse it.

Reply:
Please check for similar points throughout your code.

Reply:
Reviewed the tests and moved common mock data outside of test cases.